### PR TITLE
fix(DB/Zone):improved Zangamarsh

### PR DESCRIPTION
**Pull request by Knindzagxg**
## Changes Proposed:
- Missing SAI creature updates.
- Improvements to existing if possible.

## Tests Performed:
- Applied SQL without problems.
- Tested only Fungal Giant (19734) due extra spawn during combat.

## How to Test:
- Apply SQL(s).
- Restart Server (You can .reload creature_template for each creature as well, but will take time)
- Go to zone (Check PR name) and check creatures around - You can also spawn (.npc add) each by command.

## Source for Changes:
- [Wowhead](https://wowhead.com) Database and or comments.
- [WotlkDB](https://wotlkdb.com/) Multi checking trough Private DB's.
- [WowWiki](https://wowwiki-archive.fandom.com/wiki/Portal:Main) Can be useful for extra informations.
- [WowPedia](https://wow.gamepedia.com/Wowpedia) Cross checking trough comments if possible.
- [Retail](https://worldofwarcraft.com/en-gb/) In-game checking if previous don't have any required informations.

## Target Branch:
- [x] Master

## Operating System:
- Linux (Debian 10)

## How to Test AzerothCore PR's
When a PR is ready to be tested, it will be marked as **[Waiting to be tested]**.
You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:
http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by pklloveyou** [comment]
First of all, it's better to have AI than none.

File: data/sql/updates/pending_db_world/rev_1616252380101739000.sql [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+(20079, 0, 0, 0, 2, 0, 100, 1, 5, 20, 0, 0, 25, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Darkcrest Sentry - Between 5-20% Health - Flee For Assist (No Repeat)');
+
+UPDATE `creature_template` SET `AIName` = 'SmartAI' WHERE `entry` = 18122;
+DELETE FROM `smart_scripts` WHERE (source_type = 0 AND entryorguid = 18122);
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
+(18122, 0, 0, 1, 4, 0, 100, 1, 0, 0, 0, 0, 11, 41363, 0, 0, 0, 0, 0, 19, 18122, 5, 0, 0, 0, 0, 0, 'Dreghood Drudge - On Aggro - Cast \'41363\' (No Repeat)');
```
Comment (pklloveyou) [inline data/sql/updates/pending_db_world/rev_1616252380101739000.sql:60]:
```sql
(18122, 0, 0, 0, 4, 0, 100, 1, 0, 0, 0, 0, 11, 41363, 0, 0, 0, 0, 0, 19, 18122, 5, 0, 0, 0, 0, 0, 'Dreghood Drudge - On Aggro - Cast \'41363\' (No Repeat)');

```

Comment (Knindzagxg) [reply data/sql/updates/pending_db_world/rev_1616252380101739000.sql:60]: What's wrong with this one?

Comment (pklloveyou) [reply data/sql/updates/pending_db_world/rev_1616252380101739000.sql:60]: Link 1 is empty

Comment (Knindzagxg) [reply data/sql/updates/pending_db_world/rev_1616252380101739000.sql]: Sorted.

**Comment by Eldarix** [comment]
Here are the results of my checkings :) 

WoWScrnShot_052921_062435
WoWScrnShot_052921_062653
WoWScrnShot_052921_062812
WoWScrnShot_052921_062932
WoWScrnShot_052921_062947
WoWScrnShot_052921_063101
WoWScrnShot_052921_063120
WoWScrnShot_052921_063151
WoWScrnShot_052921_063233
WoWScrnShot_052921_063259
WoWScrnShot_052921_063320
WoWScrnShot_052921_063356
WoWScrnShot_052921_063414
WoWScrnShot_052921_063451
WoWScrnShot_052921_063536
WoWScrnShot_052921_063610
WoWScrnShot_052921_063857
WoWScrnShot_052921_063914
WoWScrnShot_052921_064010
WoWScrnShot_052921_064100
WoWScrnShot_052921_064122
WoWScrnShot_052921_064151
WoWScrnShot_052921_064305
WoWScrnShot_052921_064345
WoWScrnShot_052921_064400
WoWScrnShot_052921_064659
WoWScrnShot_052921_064823
WoWScrnShot_052921_064853
WoWScrnShot_052921_064942

**Comment by Sedatyf** [comment]
Did you test all creatures? Did all the creatures do the spells we expected? And at the right time (at aggro, below 30%, etc.)

**Comment by Eldarix** [comment]
The verification of my results are in the screenshots, I don't know if any npc escaped me because there were many but I would say yes, and obviously I focused on seeing that the expected skills get casted at the indicated life ranges :)

Thx and GL.

**Comment by Sedatyf** [comment]
>  even so I'm not sure if the skills execution has been completely correct

What do you mean?

**Comment by Sedatyf** [comment]
And I checked, and you didn't forget any creatures

**Comment by Eldarix** [comment]
I mean, I'm not sure if, for example, some skill wasn't triggered when it should because it has a % and it's not always a must :D

think that as I have said on some occasion, many of you who participate and collaborate I see that you have a minimum knowledge in these environments that I lack: D and that is why I admire you and i'm learning a lot !!

Any advice is well appreciated, take care pal.

**Comment by Sedatyf** [comment]
Haha you're too kind @Eldarix 
For those PR, what I usually do, is checking what is said in the SQL file. For example:
```SQL
<!--StartFragment-->UPDATE `creature_template` SET `AIName` = 'SmartAI' WHERE `entry` = 20293;DELETE FROM `smart_scripts` 
WHERE (source_type = 0 AND entryorguid = 20293);INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES(20293, 0, 0, 0, 4, 0, 100, 1, 0, 0, 0, 0, 11, 3248, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Bogstrok Clacker - On Aggro - Cast \'3248\' (No Repeat)');<!--EndFragment-->
```
In this part of the SQL file, at the end, you have this ``'Bogstrok Clacker - On Aggro - Cast \'3248\' (No Repeat)');```
It means that this creature Bogstrok Clacker will cast the spell ``3248`` on aggro. No repeat means that it won't cast it again.
So you have to check on a database like wowhead (https://www.wowhead.com/spell=3248) the name of the spell. And then, you check if it casts it. 
You'll have to use your jugement as well, because sometime, it can be the wrong spell id, and it makes no sense that this kind of creature casts this kind of spell.

Is this clear enough?

**Comment by Eldarix** [comment]
Honestly, it's not false modesty, it's simply that I have known the game for a long time and had never crossed the barrier of beginning to understand as much about how it works from behind, and that the emulation is reverse engineering, amazes me.

So if I understood correctly, in cases like this it is not necessary to go to check it 'live' in the game, but to apply the query in heidi and then check that it is right, no?

**Comment by Sedatyf** [comment]
No you have to apply SQL queries on your database, then check in game that creature behaviors match with what @Knindzagxg wrote in the database. But you can't have blind trust in developers (it's not personal, it's tester work) so you have to check if what @Knindzagxg wrote in the database is correct.

For example, if you want to check the first creature:
1. you check what we have in the database. It says that it should cast https://www.wowhead.com/spell=3248 on aggro
2. you check the creature data to see if it's the correct spell. If there is no information about it, think logically about it
3. you check in game if the right spell is cast, and then if it has been cast on the right time (meaning on aggro, or below 50% or whatever)

If everything pass that, you can say that it's okay. You can check my previous test on outlands area if you want to know more: https://github.com/azerothcore/azerothcore-wotlk/pull/6109#issuecomment-850836677

**Comment by Eldarix** [comment]
> No you have to apply SQL queries on your database, then check in game that creature behaviors match with what @Knindzagxg wrote in the database.

I think in this point we are referring to the same thing :D 

Screenshot 2021-05-29 183809

I imagine there are more tools for the same functionality and not just heidi, and that's why you didn't understand me, at least for changes only in DB this is how I learned to do it thanks to Branel who helped me with my first attempt.

So you confirm to me what in general terms I already had in mind about understanding how to do good tests; almost always we will have to enter the game and verify that something really works, no matter how much we see data tables or not, and that makes me happy, otherwise it would not feel so real that we are testers !! 

Thanks again for the explanations tho.

**Review by Helias — APPROVED** [review]

**Pull request merged**